server/integration/event-logger/src/main/java/org/infinispan/server/eventlogger/ServerEventImpl.java

```diff
@@ -20,17 +14,24 @@
  * @author Tristan Tarrant
  * @since 8.2
  */
-@SerializeWith(ServerEventImpl.Externalizer.class)
 public class ServerEventImpl implements EventLog {
 
-   private final EventLogLevel level;
-   private final EventLogCategory category;
-   private final Instant when;
-   private final String message;
-   private final Optional<String> detail;
-   private final Optional<String> context;
-   private final Optional<String> who;
-   private final Optional<String> scope;
+   @ProtoField(number = 1, required = true)
+   EventLogLevel level;
+
+   @ProtoField(number = 2, required = true)
+   EventLogCategory category;
+
+   @ProtoField(number = 3, required = true)
+   String message;
+
+   Instant when;
+   Optional<String> detail;
+   Optional<String> context;
+   Optional<String> who;
+   Optional<String> scope;
```

Comment:
IntelliJ warns about Optional fields anyway, so maybe it's better to keep these fields regular strings, move the annotations here, and make the methods return `Optional.ofNullable()`